### PR TITLE
Average goals per game

### PR DESCRIPTION
**Pull request by nicorithner**
Added test and methods for average_goals_per_game method as required by iteration 2 game statistics

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -54,5 +54,11 @@ def lowest_total_score
     output.total_game_score
   end
 
+  def average_goals_per_game
+    games_count = @games.count.to_f
+    sum_of_goals = (@games.map {|game| game.total_game_score}.to_a).sum
 
-end
+    sum_of_goals_divided_by_game_count = (sum_of_goals / games_count).round(2)
+    sum_of_goals_divided_by_game_count
```
Comment (Arique1104) [inline lib/stat_tracker.rb:62]: Hey Nico,

This looks great and all the test past on my end.  I guess my only question is, do we need to have `sum_of goals_divided_by_game_count` saved into a variable?

I believe this method will also pass if we simply  have 
```ruby
(sum_of_goals / games_count).round(2)
```

As the last line?

File: test/stat_tracker_test.rb [hunk 2]
```diff
@@ -31,4 +31,9 @@ def test_it_can_calculate_the_lowest_total_score
 
   end
 
+  def test_average_goals_per_game
+    # binding.pry
+    assert_equal 4.22, @stat_tracker.average_goals_per_game
```
Comment (Arique1104) [inline test/stat_tracker_test.rb:36]: Your tests are phenomenal, I am super glad we have some of the expected results we need handy.  That helps us guide our work and I'm really appreciative that you found this work around.

**Review by Arique1104 — CHANGES_REQUESTED** [review]
Hey Nico,

Overall, this works, it's in green.  I do think that we can make this method more performant by removing some of the variable usages towards the end of the method.

File: lib/stat_tracker.rb (hunk 1 above)
Comment (nicorithner) [reply lib/stat_tracker.rb:62]: done

**Comment by Arique1104** [comment]
This is not okay.  I was the reviewer for this method.  You were out of pocket Daniel for merging this method when it was assigned to me.

**Pull request merged**